File: src/main/java/com/example/blog/service/CommentService.java
package com.example.blog.service;

import com.example.blog.entity.PostEntity;
import com.example.blog.repository.CommentRepository;
import com.example.blog.entity.CommentEntity;

import com.example.blog.repository.PostRepository;
import com.google.common.base.Preconditions;
import org.springframework.stereotype.Service;

@Service
public class CommentService {
    private final CommentRepository commentRepository;
    private final PostRepository postRepository;

    CommentService(CommentRepository commentRepository, PostRepository postRepository) {
        this.commentRepository = commentRepository;
        this.postRepository = postRepository;
    }

    public void addComment(Long accountId, Long postId, String content) {
        PostEntity post = postRepository.findById(postId).orElseThrow(NullPointerException::new);
        Preconditions.checkState(accountId == post.getAccountId(),"Authorization Failed");
        commentRepository.save(new CommentEntity(accountId, postId, content));
    }
}

File: src/main/kotlin/com/example/blog/service/CommentService.kt
package com.example.blog.service

import com.example.blog.dto.CommentDto
import com.example.blog.dto.CreateCommentRequest
import com.example.blog.entity.Comment
import com.example.blog.repository.CommentRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class CommentService(
    private val commentRepository: CommentRepository
) {
    fun addComment(accountId: Long, accountName: String, request: CreateCommentRequest): CommentDto {
        return commentRepository.save(accountId, accountName, request).toDto()
    }

    private fun Comment.toDto(): CommentDto {
        return CommentDto(
            id.value,
            postId.value,
            accountName,
            content,
            createdAt,
            updatedAt
        )
    }
}